client/components/app.jsx

```diff
@@ -0,0 +1,532 @@
+        }
+      }
+
+      if (found) {
+        if (currentStars.length === 1) {
+          this.setState({
+            currentStars: [5, 4, 3, 2, 1],
+            sortingByStars: false
+          }, () => {
+            this.determineSort()})
+        } else {
+          for (var i = 0; i < currentStars.length; i++) {
+            if (currentStars[i] === 3) {
+              currentStars.splice(i, 1);
+            }
+          }
+          this.setState({
+            currentStars: currentStars
+          }, () => {this.determineSort()});
+        }
+      } else {
+        currentStars.push(3);
+        this.setState({
+          currentStars: currentStars
+        }, () => {this.determineSort()});
+      }
+    }
+  }
+
+  sortBy2() {
+    if (!this.state.sortingByStars) {
+      this.setState({
+        currentStars: [2],
+        sortingByStars: true
+      }, () => {this.determineSort()});
+    } else {
+      var currentStars = this.state.currentStars;
+      var found = false;
+      for (var i = 0; i < currentStars.length; i++) {
+        if (currentStars[i] === 2) {
+          found = true;
+        }
+      }
+
+      if (found) {
+        if (currentStars.length === 1) {
+          this.setState({
+            currentStars: [5, 4, 3, 2, 1],
+            sortingByStars: false
+          }, () => {
+            this.determineSort()})
+        } else {
+          for (var i = 0; i < currentStars.length; i++) {
+            if (currentStars[i] === 2) {
+              currentStars.splice(i, 1);
+            }
+          }
+          this.setState({
+            currentStars: currentStars
+          }, () => {this.determineSort()});
+        }
+      } else {
+        currentStars.push(2);
+        this.setState({
+          currentStars: currentStars
+        }, () => {this.determineSort()});
+      }
+    }
+  }
+
+  sortBy1() {
+    if (!this.state.sortingByStars) {
+      this.setState({
+        currentStars: [1],
+        sortingByStars: true
+      }, () => {this.determineSort()});
+    } else {
+      var currentStars = this.state.currentStars;
+      var found = false;
+      for (var i = 0; i < currentStars.length; i++) {
+        if (currentStars[i] === 1) {
+          found = true;
+        }
+      }
+
+      if (found) {
+        if (currentStars.length === 1) {
+          this.setState({
+            currentStars: [5, 4, 3, 2, 1],
+            sortingByStars: false
+          }, () => {
+            this.determineSort()})
+        } else {
+          for (var i = 0; i < currentStars.length; i++) {
+            if (currentStars[i] === 1) {
+              currentStars.splice(i, 1);
+            }
+          }
+          this.setState({
+            currentStars: currentStars
+          }, () => {this.determineSort()});
+        }
+      } else {
+        currentStars.push(1);
+        this.setState({
+          currentStars: currentStars
+        }, () => {this.determineSort()});
+      }
+    }
+  }
+
```

Comment:
All sorting functions look very similar. Creating a helper function for sorting by star and then calling it for each different type of star count will help follow the DRY principle.